src/workerd/io/actor-sqlite.c++

```diff
@@ -546,7 +563,6 @@ void ActorSqlite::shutdown(kj::Maybe<const kj::Exception&> maybeException) {
 
 kj::OneOf<ActorSqlite::CancelAlarmHandler, ActorSqlite::RunAlarmHandler> ActorSqlite::
     armAlarmHandler(kj::Date scheduledTime, bool noCache) {
-  KJ_ASSERT(!haveDeferredDelete);
```

Comment:
Why did this need to be removed? I'd still think it should always be false when starting a new alarm handler?

Reply:
I wasn't sure that's the case...  I think we only currently set it to false when:

- setAlarm() is called on the root level transaction
- committing a transaction with a dirty alarm value
- starting a new alarm handler when the handler time is later than the local database state

We're not currently setting it to false when a DeferredAlarmDeleter is destroyed and starts the deferred alarm deletion, and maybe we should be?  But I'm also not totally sure if it's possible for the DeferredAlarmDeleter to be attached to something that takes a few turns of the event loop to fully resolve, such that it's possible for other work in the event queue to start a new handler before DeferredAlarmDeleter's destructor runs.

But thinking about it more, yeah, this seems unlikely, and I think should already be covered by the log-once warnings, so I'll try reintroducing the assert and fix the DeferredAlarmDeleter to clear the flag.

Reply:
Readded the check as a logged-once warning and added code to clear the flag, in cb67b25 .